packages/components/upload/src/upload.vue

```diff
@@ -114,4 +100,49 @@ defineExpose({
   /** @description remove the file manually */
   handleRemove,
 })
+
+const slots = useSlots()
+
+const UploadContentWithProps = defineComponent({
+  render() {
+    return h(
+      UploadContent as any,
+      {
+        ref: 'uploadRef',
+        type: props.type,
+        drag: props.drag,
+        action: props.action,
+        multiple: props.multiple,
+        withCredentials: props.withCredentials,
+        headers: props.headers,
+        method: props.method,
+        name: props.name,
+        data: props.data,
+        accept: props.accept,
+        autoUpload: props.autoUpload,
+        listType: props.listType,
+        disabled: props.disabled,
+        limit: props.limit,
+        fileList: props.fileList,
+        httpRequest: props.httpRequest,
+        beforeUpload: props.beforeUpload,
+        onExceed: props.onExceed,
+        onStart: handleStart,
+        onProgress: handleProgress,
+        onSuccess: handleSuccess,
+        onError: handleError,
+        onRemove: handleRemove,
+      },
+      {
+        default: () => {
+          if (slots.trigger) {
```

Comment:
Is this equivalent to something like:

```suggestion
<UploadContent v-bind="props">
  <slot name="trigger" v-if="slots.trigger"/>
  <slot v-if="slots.default"/>
</UploadContent>
```

Reply:
But not all props were passed in `UploadContent`.

If use template, I have to maintain it in two places.

Reply:
But creating a component on the fly has undesirable performance impact. For UI framework,  we should prefer end users' performance over slight duplication. Especially the burden isn't large.

Reply:
Ok. I rewrite it with `template`.